src/main/java/io/specto/hoverfly/junit/core/Hoverfly.java

```diff
@@ -265,6 +281,13 @@ public void simulate(SimulationSource simulationSource, SimulationSource... sour
                         simulation.getHoverflyData().getGlobalActions().getDelays().addAll(s.getHoverflyData().getGlobalActions().getDelays());
                     });
 
+            if (isSimulationPreprocessingEnabled(testInstance)) {
+                Optional<SimulationPreprocessorProvider> simulationPreprocessorProvider = hoverflyConfig.getSimulationPreprocessorProvider();
```

Comment:
I think the declaration of this optional variable is not necessary, you can just call `ifPresent` after get.